Week3/homework/step2-7.js

```diff
@@ -12,12 +12,14 @@ console.log(x);
 
 const y = { x: 9 };
 function f2(val) {
-  val.x = val.x + 1;
-  return val;
+	val.x = val.x + 1;
+	return val;
 }
 
 f2(y);
 
 console.log(y);
 
 // Add your explanation as a comment here
+// first example returns 9: x is a constant in the first example and cant change its value? 
+//second example is {x: 10}: Constants cant be changed, but in this case it is an object and arrays and objects can be change when assigned to a constant.
```

Comment:
You are close. Some primitives are `pass by value`...

```
let a = 4
let b = a
a += 1
What's the value of b?
```
vs objects/arrays which are `pass by reference`...
```
let a = { name: 'gabe' }
b = a
a.country = 'us'
what's the value of b?
```
Do you see the difference?

Reply:
https://www.mathwarehouse.com/programming/images/pass-by-reference-vs-pass-by-value-animation.gif
Maybe? I am kinda bad understanding this, would b pass by value still be 4? and then b by reference name gabe country us?

Reply:
yes!